src/plugins/discourse/mirror.js

```diff
@@ -67,15 +73,109 @@ export class Mirror {
       ...defaultOptions,
       ...(options || {}),
     };
+    this._mode = mode || 1;
   }
 
   async update(reporter: TaskReporter) {
     reporter.start("discourse");
-    await this._updateTopics(reporter);
-    await this._updateLikes(reporter);
+    switch (this._mode) {
+      case 1:
+        await this._updateTopics(reporter);
+        await this._updateLikes(reporter);
+        break;
+      case 2:
+        await this._updateTopicsV2(reporter);
+        await this._updateLikes(reporter);
+        break;
+      default:
+        throw new Error(`Invalid mode ${this._mode}, expecting 1 or 2`);
```

Comment:
If we switch mode to an enum, then we can do ```throw new Error(`Invalid mode: ${this._mode: empty}`)``` and then flow will correctly type error if we ever add a new mode, but don't handle it.

Reply:
Changed to an enum.